examples/postgres/axum-social-with-tests/src/http/mod.rs

```diff
@@ -15,12 +16,12 @@ pub fn app(db: PgPool) -> Router {
     Router::new()
         .merge(user::router())
         .merge(post::router())
-        .layer(Extension(db))
+        .with_state(db)
```

Comment:
What's actually a more useful pattern here is to define a new `AppState` struct that holds everything your routes might need:

```rust
#[derive(axum::FromRef)]
struct AppState {
    db: PgPool,
}
```

Then all `router()` methods return `Router<AppState>`.

The request handlers don't actually have to change at all because the `State` extractor automatically extracts anything that implements `FromRef<AppState>`.

https://docs.rs/axum/latest/axum/extract/struct.State.html#substates